### ISSUE TITLE
Code Insights: Adopt separate insights loading system for the homepage and directory page.

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#23188]
### Requirements 
Migration from the extension API should be resolved to be able to start this issue https://github.com/sourcegraph/sourcegraph/issues/23187

### Context

Currently, we have 3 different places for insights in the main Sourcegraph app. 
- Insight page (supports only search-based, code-stats, and BE insights) 
- Homepage supports only search-based and BE insights
- Directory page supports all extension insights (search-based, code-stats, other insight extensions from the extension store like codecov) and BE insights

With the new loading system which this ticket implies https://github.com/sourcegraph/sourcegraph/issues/23187 we should adopt the new loading system for the homepage and directory page as well to be ably to support all types of insights (BE insights, search-based built-in, code-stats built-in let's call them FE insights, and extension based insights) at any place from the list above. 

For that, we have to implement reusable logic or reusable components for the loading and viewing all these 3 types of insight in these places (BE insights, FE insights - built-in search and code stats insights, extension-based insights)

**Comment by Joelkw** [comment]
@vovakulikov I'm not sure if this is worth prioritizing for the beta – does it only affect the homepage/directory page views and not the dashboard view? In which case, I would be in favor of just pulling them out / disabling them by default for now and giving us this time to polish/pad our beta freeze cushion. (And we can return to this once we have a clearer defined vision for the homepage and directory page views.)

**Comment by vovakulikov** [comment]
@Joelkw I think you're right about this. I think this PR does enough for the beta stage (I mean insight loading improvements) https://github.com/sourcegraph/sourcegraph/pull/23244

We can skip this ticket and this one https://github.com/sourcegraph/sourcegraph/issues/23187 for now and take a look at these later.

**Comment by felixfbecker** [comment]
I agree with @Joelkw, removing from drilldown project since it seems we have consensus.

I also want to point out that what the issue description says is not correct, unless something was changed:

> Insight page (supports only search-based, code-stats, and BE insights)
> Homepage supports only search-based and BE insights

The insights page and homepage support extension-based insights just like the directory page (extensions can choose to register view providers for those pages). It's just that no extension we have commonly enabled happens to do so (e.g. the Codecov API only gives us data for a repo, so we can't display anything in a global context). That doesn't mean it's not _supported_ on those pages though, as the issue description claims. Afaik this goes through the same loading functions (at least it used to) which would also query for extension-contributed views.
